### PR TITLE
Bump mariadb connector from 3.0.3 -> 3.1.9

### PR DESCRIPTION
**Pull request by peak3d**
## Description
Bump connector version from 3.0.3 to latest released 3.1.9

## Motivation and Context
At least for Android 3.0.3 does not work properly with newer server versions (tested 10.5.4)
Field types in resultsets are not returned correctly.

## How Has This Been Tested?
NVIDIA Shield TV, Ubuntui 18.04 with mariadb server 10.5.4.
- with 3.0.3 fail (see sceenshot), with 3.1.9 success.

## Screenshots (if appropriate):
image

## Types of change
- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

### CODE REVIEW

**Comment by fuzzard** [comment]
Jenkins build this please

**Comment by peak3d** [comment]
Jenkins build this please

File: tools/depends/target/mariadb/01-android.patch [hunk 1]
```diff
@@ -1,31 +1,36 @@
+diff --git a/cmake/CheckIncludeFiles.cmake b/cmake/CheckIncludeFiles.cmake
+index ed71adc..50a9a49 100644
```
Comment (Rechi) [inline tools/depends/target/mariadb/01-android.patch:2]:
```suggestion
```

File: tools/depends/target/mariadb/01-android.patch [hunk 2]
```diff
@@ -1,31 +1,36 @@
...
 --- a/cmake/CheckIncludeFiles.cmake
 +++ b/cmake/CheckIncludeFiles.cmake
-@@ -74,3 +74,12 @@
- CHECK_INCLUDE_FILES (unistd.h HAVE_UNISTD_H)
- CHECK_INCLUDE_FILES (utime.h HAVE_UTIME_H)
- CHECK_INCLUDE_FILES (ucontext.h HAVE_UCONTEXT_H)
-+IF(HAVE_UCONTEXT_H)
+@@ -50,4 +50,11 @@ CHECK_INCLUDE_FILES (utime.h HAVE_UTIME_H)
```
Comment (Rechi) [inline tools/depends/target/mariadb/01-android.patch:5]:
```suggestion
@@ -50,4 +50,11 @@
```

File: tools/depends/target/mariadb/01-android.patch [hunk 3]
```diff
@@ -1,31 +1,36 @@
...
 +  CHECK_FUNCTION_EXISTS (makecontext HAVE_MAKECONTEXT)
 +  CHECK_FUNCTION_EXISTS (setcontext HAVE_SETCONTEXT)
 +  CHECK_FUNCTION_EXISTS (swapcontext HAVE_SWAPCONTEXT)
 +  IF(NOT HAVE_GETCONTEXT OR NOT HAVE_MAKECONTEXT OR NOT HAVE_SETCONTEXT OR NOT HAVE_SWAPCONTEXT)
 +    SET(HAVE_UCONTEXT_H 0 CACHE INTERNAL "")
 +  ENDIF()
-+ENDIF(HAVE_UCONTEXT_H)
+ ENDIF()
+diff --git a/cmake/CheckTypes.cmake b/cmake/CheckTypes.cmake
+index 2a47cc0..01759dc 100644
```
Comment (Rechi) [inline tools/depends/target/mariadb/01-android.patch:18]:
```suggestion
```

File: tools/depends/target/mariadb/01-android.patch [hunk 4]
```diff
@@ -1,31 +1,36 @@
...
 --- a/cmake/CheckTypes.cmake
 +++ b/cmake/CheckTypes.cmake
-@@ -25,6 +25,7 @@
- CHECK_TYPE_SIZE(off_t SIZEOF_OFF_T)
+@@ -22,6 +22,7 @@ CHECK_TYPE_SIZE(size_t SIZEOF_SIZE_T)
```
Comment (Rechi) [inline tools/depends/target/mariadb/01-android.patch:21]:
```suggestion
@@ -22,6 +22,7 @@
```

File: tools/depends/target/mariadb/01-android.patch [hunk 5]
```diff
@@ -1,31 +1,36 @@
...
+ SET(CMAKE_EXTRA_INCLUDE_FILES sys/types.h)
  CHECK_TYPE_SIZE(uchar SIZEOF_UCHAR)
  CHECK_TYPE_SIZE(uint SIZEOF_UINT)
 +CHECK_TYPE_SIZE(ushort SIZEOF_USHORT)
  CHECK_TYPE_SIZE(ulong SIZEOF_ULONG)
  CHECK_TYPE_SIZE(int8 SIZEOF_INT8)
  CHECK_TYPE_SIZE(uint8 SIZEOF_UINT8)
+diff --git a/include/ma_config.h.in b/include/ma_config.h.in
+index 1e57257..ba6d6be 100644
```
Comment (Rechi) [inline tools/depends/target/mariadb/01-android.patch:30]:
```suggestion
```

File: tools/depends/target/mariadb/01-android.patch [hunk 6]
```diff
@@ -35,11 +40,13 @@
 +#endif
 +
  #cmakedefine SIZEOF_ULONG @SIZEOF_ULONG@
- #if SIZEOF_ULONG
+ #if defined(SIZEOF_ULONG)
  # define HAVE_ULONG 1
+diff --git a/include/ma_global.h b/include/ma_global.h
+index ee74ffa..9873e98 100644
```
Comment (Rechi) [inline tools/depends/target/mariadb/01-android.patch:46]:
```suggestion
```

File: tools/depends/target/mariadb/01-android.patch [hunk 7]
```diff
@@ -35,11 +40,13 @@
...
 --- a/include/ma_global.h
 +++ b/include/ma_global.h
-@@ -272,6 +272,8 @@
+@@ -268,6 +268,8 @@ double my_ulonglong2double(unsigned long long A);
```
Comment (Rechi) [inline tools/depends/target/mariadb/01-android.patch:49]:
```suggestion
@@ -268,6 +268,8 @@
```

File: tools/depends/target/mariadb/01-android.patch [hunk 8]
```diff
@@ -48,9 +55,11 @@
  typedef unsigned short ushort;
  #endif
  
+diff --git a/libmariadb/CMakeLists.txt b/libmariadb/CMakeLists.txt
+index f406b37..43f7b4f 100644
```
Comment (Rechi) [inline tools/depends/target/mariadb/01-android.patch:59]:
```suggestion
```

File: tools/depends/target/mariadb/01-android.patch [hunk 9]
```diff
@@ -48,9 +55,11 @@
...
 --- a/libmariadb/CMakeLists.txt
 +++ b/libmariadb/CMakeLists.txt
-@@ -291,7 +291,6 @@
+@@ -311,7 +311,6 @@ ELSE()
```
Comment (Rechi) [inline tools/depends/target/mariadb/01-android.patch:62]:
```suggestion
@@ -311,7 +311,6 @@
```

**Review by Rechi — COMMENTED** [review]
Please follow the common patch style used for all depends.

**Comment by peak3d** [comment]
@Rechi the diff was automatically generated by git diff - is there any option I have to pass git diff do generate a patch file with the changes you suggested?

**Comment by fuzzard** [comment]
Jenkins build this please

**Comment by skylex** [comment]
I've got the same issue on Windows 11 and Kodi (19.3 (19.3.0) Git:20211024-49a04cd6a7) when using my external MariaDB videodatabase: every item is labeled as '0' in the recently added movies section, but on some places like the homescreen I can see the movies names...
Very strange behavior and I did not found any error in the debug log and nothing else that could point to the problem.

The same version of Kodi on Debian 11 using the same MariaDB database is not affected.

Anyone else having this issue on W11 (Pro x64) ?

**Comment by lancercncs1** [comment]
Hmm.. I am having similar issues with my iMac kodi client running Monterey. MySql/MariaDB syncing works just fine on my Linux/Rpi4 clients, but not my macOS one. Any idea how to resolve this on macOS?

**Comment by lrusak** [comment]
@lancercncs1 please use the forum for support. https://forum.kodi.tv/

**Pull request merged**